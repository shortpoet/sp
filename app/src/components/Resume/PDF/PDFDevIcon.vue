<template>
<div class="">
<i v-if="!isImage" :class="source"></i>
<img v-else :class="imgClass" :src="source" />
<span v-if="showName" :class="devicon - name">{{ name }}</span>
</div>
</template>

<script>
export default {
name: "PDFDevIcon",
props: {
source: {
type: String,
},
name: {
type: String,
default: null,
},
showName: {
type: Boolean,
default: false,
},
},
components: {},
data() {
return {};
},
computed: {
isImage() {
return this.source.match(/svg|jpg|png|jpeg/g) !== null;
},
imgClass() {
let imgClass = "";
switch (this.name) {
case "vue":
imgClass = "vue-devicon";
break;
case "ts":
imgClass = "ts-devicon";
break;
case "tableau":
imgClass = "tableau-devicon";
break;
case "d3":
imgClass = "d3-devicon";
break;
case "azure":
imgClass = "azure-devicon";
break;
case "bash":
imgClass = "bash-devicon";
break;
case "terraform":
imgClass = "terraform-devicon";
break;
case "powershell":
imgClass = "powershell-devicon";
break;
case "aws":
imgClass = "aws-devicon";
break;
}
return imgClass;
},
},
methods: {},
mounted() {},
};
</script>
