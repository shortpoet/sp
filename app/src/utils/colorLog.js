const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
