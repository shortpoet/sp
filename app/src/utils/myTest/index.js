import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
