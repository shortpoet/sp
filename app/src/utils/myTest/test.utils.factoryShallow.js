import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow