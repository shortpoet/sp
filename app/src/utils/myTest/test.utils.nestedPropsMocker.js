

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
