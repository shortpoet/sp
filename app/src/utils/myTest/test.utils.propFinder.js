import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder