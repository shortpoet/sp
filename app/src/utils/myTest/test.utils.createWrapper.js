import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper