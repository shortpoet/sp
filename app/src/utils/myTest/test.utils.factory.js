import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory