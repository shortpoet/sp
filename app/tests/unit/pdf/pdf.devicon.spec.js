import * as myTest from '@/utils/myTest'
const { createWrapper } = myTest.default
import PDFDevIcon from '@/components/Resume/PDF/PDFDevIcon'
import { cloneDeep } from 'lodash'

describe('PDFDevIcon.vue', () => {

  const component = PDFDevIcon
  const mountOptions = {
    propsData: {
      source: 'sample_source',
      name: '',
      showName: false
    }
  }

  let wrapper

  let expected
  let isImage
  let imgClass

  const imageExtensions = ['.svg', '.jpg', '.png', '.jpeg']
  const names = ['vue', 'tableau', 'd3', 'ts', 'azure', 'bash', 'powershell', 'terraform', 'aws']

  beforeEach(() => {
    isImage = false
    imgClass = ''
    expected = ''
    jest.resetModules()
    jest.clearAllMocks()
  })

  describe('renders i if not isImage with class containing source', () => {

    const _mountOptions = cloneDeep(mountOptions)
    _mountOptions.propsData.source += ``
    expected += ``

    it('renders i if not isImage with class containing source', () => {

      wrapper = createWrapper(component, _mountOptions)

      expect(wrapper.find('i').attributes().class).toContain(expected)

    })

    it('matches snapshot', () => {

      wrapper = createWrapper(component, _mountOptions)

      expect(wrapper.html()).toMatchSnapshot()

    })

  })


  imageExtensions.forEach(ext => {
    names.forEach(name => {
      describe(`renders img if isImage with class containing source ${ext} and name ${name}`, () => {

        const _mountOptions = cloneDeep(mountOptions)

        _mountOptions.propsData.source += ext

        _mountOptions.propsData.name += name

        wrapper = createWrapper(component, _mountOptions)

        switch (name) {
          case "vue":
            imgClass = "vue-devicon";
            break;
          case "ts":
            imgClass = "ts-devicon";
            break;
          case "tableau":
            imgClass = "tableau-devicon";
            break;
          case "d3":
            imgClass = "d3-devicon";
            break;
          case "azure":
            imgClass = "azure-devicon";
            break;
          case "bash":
            imgClass = "bash-devicon";
            break;
          case "terraform":
            imgClass = "terraform-devicon";
            break;
          case "powershell":
            imgClass = "powershell-devicon";
            break;
          case "aws":
            imgClass = "aws-devicon";
            break;
        }

        expected += `${imgClass}`

        it(`renders img if isImage with class containing source ${ext} and name ${name}`, () => {

          wrapper = createWrapper(component, _mountOptions)

          expect(wrapper.find('img').attributes().class).toContain(expected)

        })

        it('matches snapshot', () => {

          wrapper = createWrapper(component, _mountOptions)

          expect(wrapper.html()).toMatchSnapshot()

        })
        it('renders span with name if showName true', () => {

          const __mountOptions = cloneDeep(_mountOptions)

          __mountOptions.propsData.showName = true

          wrapper = createWrapper(component, __mountOptions)

          expect(wrapper.find('span').text()).toMatch(name)

        })
        it('matches snapshot with showname', () => {

          const __mountOptions = cloneDeep(_mountOptions)

          __mountOptions.propsData.showName = true

          wrapper = createWrapper(component, __mountOptions)

          expect(wrapper.html()).toMatchSnapshot()

        })
      })
    })

  })

})
