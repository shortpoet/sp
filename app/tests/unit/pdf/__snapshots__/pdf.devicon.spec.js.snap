// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFDevIcon.vue renders i if not isImage with class containing source matches snapshot 1`] = `
<div><i class="sample_source"></i>
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name aws matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="aws-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name aws matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="aws-devicon"> <span class="">aws</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name azure matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="azure-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name azure matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="azure-devicon"> <span class="">azure</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name bash matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="bash-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name bash matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="bash-devicon"> <span class="">bash</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name d3 matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="d3-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name d3 matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="d3-devicon"> <span class="">d3</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name powershell matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="powershell-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name powershell matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="powershell-devicon"> <span class="">powershell</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name tableau matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="tableau-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name tableau matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="tableau-devicon"> <span class="">tableau</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name terraform matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="terraform-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name terraform matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="terraform-devicon"> <span class="">terraform</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name ts matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="ts-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name ts matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="ts-devicon"> <span class="">ts</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name vue matches snapshot 1`] = `
<div><img src="sample_source.jpeg" class="vue-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpeg and name vue matches snapshot with showname 1`] = `<div><img src="sample_source.jpeg" class="vue-devicon"> <span class="">vue</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name aws matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="aws-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name aws matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="aws-devicon"> <span class="">aws</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name azure matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="azure-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name azure matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="azure-devicon"> <span class="">azure</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name bash matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="bash-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name bash matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="bash-devicon"> <span class="">bash</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name d3 matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="d3-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name d3 matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="d3-devicon"> <span class="">d3</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name powershell matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="powershell-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name powershell matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="powershell-devicon"> <span class="">powershell</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name tableau matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="tableau-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name tableau matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="tableau-devicon"> <span class="">tableau</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name terraform matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="terraform-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name terraform matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="terraform-devicon"> <span class="">terraform</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name ts matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="ts-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name ts matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="ts-devicon"> <span class="">ts</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name vue matches snapshot 1`] = `
<div><img src="sample_source.jpg" class="vue-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .jpg and name vue matches snapshot with showname 1`] = `<div><img src="sample_source.jpg" class="vue-devicon"> <span class="">vue</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name aws matches snapshot 1`] = `
<div><img src="sample_source.png" class="aws-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name aws matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="aws-devicon"> <span class="">aws</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name azure matches snapshot 1`] = `
<div><img src="sample_source.png" class="azure-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name azure matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="azure-devicon"> <span class="">azure</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name bash matches snapshot 1`] = `
<div><img src="sample_source.png" class="bash-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name bash matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="bash-devicon"> <span class="">bash</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name d3 matches snapshot 1`] = `
<div><img src="sample_source.png" class="d3-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name d3 matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="d3-devicon"> <span class="">d3</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name powershell matches snapshot 1`] = `
<div><img src="sample_source.png" class="powershell-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name powershell matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="powershell-devicon"> <span class="">powershell</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name tableau matches snapshot 1`] = `
<div><img src="sample_source.png" class="tableau-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name tableau matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="tableau-devicon"> <span class="">tableau</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name terraform matches snapshot 1`] = `
<div><img src="sample_source.png" class="terraform-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name terraform matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="terraform-devicon"> <span class="">terraform</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name ts matches snapshot 1`] = `
<div><img src="sample_source.png" class="ts-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name ts matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="ts-devicon"> <span class="">ts</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name vue matches snapshot 1`] = `
<div><img src="sample_source.png" class="vue-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .png and name vue matches snapshot with showname 1`] = `<div><img src="sample_source.png" class="vue-devicon"> <span class="">vue</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name aws matches snapshot 1`] = `
<div><img src="sample_source.svg" class="aws-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name aws matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="aws-devicon"> <span class="">aws</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name azure matches snapshot 1`] = `
<div><img src="sample_source.svg" class="azure-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name azure matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="azure-devicon"> <span class="">azure</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name bash matches snapshot 1`] = `
<div><img src="sample_source.svg" class="bash-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name bash matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="bash-devicon"> <span class="">bash</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name d3 matches snapshot 1`] = `
<div><img src="sample_source.svg" class="d3-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name d3 matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="d3-devicon"> <span class="">d3</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name powershell matches snapshot 1`] = `
<div><img src="sample_source.svg" class="powershell-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name powershell matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="powershell-devicon"> <span class="">powershell</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name tableau matches snapshot 1`] = `
<div><img src="sample_source.svg" class="tableau-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name tableau matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="tableau-devicon"> <span class="">tableau</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name terraform matches snapshot 1`] = `
<div><img src="sample_source.svg" class="terraform-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name terraform matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="terraform-devicon"> <span class="">terraform</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name ts matches snapshot 1`] = `
<div><img src="sample_source.svg" class="ts-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name ts matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="ts-devicon"> <span class="">ts</span></div>`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name vue matches snapshot 1`] = `
<div><img src="sample_source.svg" class="vue-devicon">
  <!---->
</div>
`;

exports[`PDFDevIcon.vue renders img if isImage with class containing source .svg and name vue matches snapshot with showname 1`] = `<div><img src="sample_source.svg" class="vue-devicon"> <span class="">vue</span></div>`;
